dome9/resource_dome9_azure_organization_onboarding.go

```diff
@@ -0,0 +1,397 @@
+	_ = d.Set("is_auto_onboarding", resp.IsAutoOnboarding)
+	_ = d.Set("update_time", resp.UpdateTime)
+	_ = d.Set("creation_time", resp.CreationTime)
+
+	//other error checks?
+
+	return nil
+}
+
+func resourceAzureOrganizationOnboardingUpdate(d *schema.ResourceData, meta interface{}) error {
+	d9Client := meta.(*Client)
+	log.Println("An update occurred")
+
+	if d.HasChange("organization_name") {
+		log.Println("The configuration has been changed")
+
+		updateConfigReq := azure_org.OnboardingUpdateRequest{
+			OrganizationName: d.Get("organization_name").(string),
+		}
+
+		if resp, err := d9Client.azureOrganizationOnboarding.UpdateOrganizationManagementAsync(d.Id(), updateConfigReq); err != nil {
+			return err
+		} else {
+			log.Printf("resourceAzureOrganizationOnboardingUpdate Configuration response is: %+v\n", resp)
+		}
+	}
+
+	return nil
+}
+
+func resourceAzureOrganizationOnboardingDelete(d *schema.ResourceData, meta interface{}) error {
+	d9Client := meta.(*Client)
+	log.Printf("[INFO] Deleting Azure organization ID: %v\n", d.Id())
+	if _, err := d9Client.azureOrganizationOnboarding.Delete(d.Id()); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func expandAzureOrganizationOnboardingRequest(d *schema.ResourceData) azure_org.OnboardingRequest {
+	req := azure_org.OnboardingRequest{
+		WorkflowId:              d.Get("workflow_id").(string),
+		TenantId:                d.Get("tenant_id").(string),
+		ManagementGroupId:       d.Get("management_group_id").(string),
+		OrganizationName:        d.Get("organization_name").(string),
+		AppRegistrationName:     d.Get("app_registration_name").(string),
+		ClientId:                d.Get("client_id").(string),
+		ClientSecret:            d.Get("client_secret").(string),
+		UseCloudGuardManagedApp: d.Get("use_cloud_guard_managed_app").(bool),
+		IsAutoOnboarding:        d.Get("is_auto_onboarding").(bool),
+		ActiveBlades: azure_org.Blades{
+			Awp: azure_org.AwpConfiguration{
+				BladeConfiguration: azure_org.BladeConfiguration{
+					IsEnabled: d.Get("active_blades.0.awp.is_enabled").(bool),
+				},
+				OnboardingMode:            azure_org.AwpOnboardingMode(d.Get("active_blades.0.awp.onboarding_mode").(string)),
+				CentralizedSubscriptionId: d.Get("active_blades.0.awp.centralized_subscription_id").(string),
+				WithFunctionAppsScan:      d.Get("active_blades.0.awp.with_function_apps_scan").(bool),
+			},
+			Serverless: azure_org.ServerlessConfiguration{
+				BladeConfiguration: azure_org.BladeConfiguration{
+					IsEnabled: d.Get("active_blades.0.serverless.is_enabled").(bool),
+				},
+			},
+			Cdr: azure_org.CdrConfiguration{
+				BladeConfiguration: azure_org.BladeConfiguration{
+					IsEnabled: d.Get("active_blades.0.cdr.is_enabled").(bool),
+				},
+				Accounts: []azure_org.StorageAccount{
+					{
+						StorageId: d.Get("active_blades.0.cdr.accounts.0.storage_id").(string),
```

Comment:
active_blades.0.
Do you think this is the correct syntax to get the relevant fields?
Pls validate it

Reply:
yes, in case of TypeList.
in this case, the right type is TypeMap and the code fixed